adapters/repos/db/inverted/objects.go

```diff
@@ -473,3 +507,43 @@ func (a *Analyzer) analyzeRefProp(prop *models.Property,
 		HasFrequency: false,
 	}, nil
 }
+
+func typedSliceToUntyped(in interface{}) ([]interface{}, error) {
+	switch typed := in.(type) {
+	case []interface{}:
+		// nothing to do
+		return typed, nil
+	case []string:
+		out := make([]interface{}, len(typed))
+		for i := range out {
+			out[i] = typed[i]
+		}
+		return out, nil
+	case []int:
+		out := make([]interface{}, len(typed))
+		for i := range out {
+			out[i] = typed[i]
+		}
+		return out, nil
+	case []time.Time:
+		out := make([]interface{}, len(typed))
+		for i := range out {
+			out[i] = typed[i]
+		}
+		return out, nil
+	case []bool:
+		out := make([]interface{}, len(typed))
+		for i := range out {
+			out[i] = typed[i]
+		}
+		return out, nil
+	case []float64:
+		out := make([]interface{}, len(typed))
+		for i := range out {
+			out[i] = typed[i]
+		}
+		return out, nil
```

Comment:
This could actually be simplified with generics:

```go
func convertToUntyped[T comparable](in []T) []interface{} {
	out := make([]interface{}, len(in))
	for i := range out {
		out[i] = in[i]
	}
	return out
}

func typedSliceToUntyped2(in any) ([]any, error) {
	switch typed := in.(type) {
	case []any:
		// nothing to do
		return typed, nil
	case []string:
		return convertToUntyped[string](typed), nil
	case []int:
		return convertToUntyped[int](typed), nil
	case []time.Time:
		return convertToUntyped[time.Time](typed), nil
	case []bool:
		return convertToUntyped[bool](typed), nil
	case []float64:
		return convertToUntyped[float64](typed), nil
	default:
		return nil, errors.Errorf("unrecognized slice type %T", in)
	}
}
```

Reply:
I have applied these changes, but with `interface{}` instead of `any` to keep it consistent with the codebase for now